FLT/DivisionAlgebra/Finiteness.lean

```diff
@@ -212,19 +309,175 @@ abbrev Dfx := (D ⊗[K] (FiniteAdeleRing (𝓞 K) K))ˣ
 noncomputable abbrev incl₁ : Dˣ →* Dfx K D :=
   Units.map Algebra.TensorProduct.includeLeftRingHom.toMonoidHom
 
+variable [MeasurableSpace (D ⊗[K] NumberField.AdeleRing (𝓞 K) K)]
+  [BorelSpace (D ⊗[K] NumberField.AdeleRing (𝓞 K) K)]
+
+def iso₁ : (D ⊗[K] NumberField.AdeleRing (𝓞 K) K)ˣ ≃*
+    Prod (D ⊗[K] NumberField.InfiniteAdeleRing K)ˣ (Dfx K D) := by
+  simp_rw [NumberField.AdeleRing, Dfx]
+  have h1 : D ⊗[K] (NumberField.InfiniteAdeleRing K × FiniteAdeleRing (𝓞 K) K) ≃
+      (D ⊗[K] (NumberField.InfiniteAdeleRing K)) × (D ⊗[K] FiniteAdeleRing (𝓞 K) K) := by
+    -- DirectSum.addEquivProd
+    -- TensorProduct.directSumRight
+
+    sorry
+  have := Units.mapEquiv h1
+  apply MulEquiv.prodUnits at this
+  sorry
+
+def rest₁ : ringHaarChar_ker D_𝔸 → Dfx K D :=
+  fun a => (iso₁ K D) a.val |>.2
+
+def α : (ringHaarChar_ker D_𝔸 ⧸ (MonoidHom.range
+    (NumberField.AdeleRing.DivisionAlgebra.incl K D)).comap (ringHaarChar_ker D_𝔸).subtype)
+    → (Dfx K D ⧸ (incl₁ K D).range) :=
+  fun a => Quot.mk _ (rest₁ K D a.out)
+
+local instance : TopologicalSpace (Dfx K D ⧸ (incl₁ K D).range) :=
+  QuotientGroup.instTopologicalSpace _
+
 theorem NumberField.FiniteAdeleRing.DivisionAlgebra.units_cocompact :
     CompactSpace (Dfx K D ⧸ (incl₁ K D).range) := by
-  sorry
+  have h1 : Continuous (α K D) := by
+    unfold α rest₁
+    refine Continuous.comp' ?_ ?_
+    · exact { isOpen_preimage := fun s a ↦ a }
+    ·
+      -- this is certainly true by isomorphism and topology on RHS -- may need to explicitely show?
```

Comment:
No, this is not true (yet). If you `simp [iso₁]` the goal reduces to  `Continuous fun a ↦ ((sorry sorry) ↑(Quotient.out a)).2` so basically this function is not even defined, so you can't prove anything about it.